src/reference/modules/reference/pages/variables.adoc

```diff
@@ -1,93 +1,404 @@
 = Variables
 
-== General
+== Generalities
 
-In Rudder, you can define and use variables via the technique editor or the differents pre-built techniques.
+Rudder has a notion of `variables`. A `variable` is something with a `name` that references a `value`, and that `value` can be accessed with appropriate syntax around its `name`. We call accessing the value `variable expansion` in honor to shell, guarding totem of devops glue. 
 
-All variables are defined under a `prefix` (scope), so to reference a variable you will always need its `prefix` and its `name`, separated via a `.` char.
-To call a variable in Rudder we use `${...}` brackets syntax as described below:
+In Rudder, `variable name` are namespaced with a dotted convention. `Values` can be maonly of two types: `string` and `json`, ie dictionnary-like recursice structures. Rudder only supports `UTF-8` for `varibles` `names` and `values`.
```

Comment:
```suggestion
In Rudder, `variable name` are namespaced with a dotted convention. `Values` can be mainly of two types: `string` and `json`, ie dictionnary-like recursive structures. Rudder only supports `UTF-8` for `variables` `names` and `values`.
```

Reply:
Actually I think variable names need to be ASCII

Reply:
the need for cfengine, but they are escaped by agent during generation. Is there an other reason to do so?

Reply:
They are put as is in properties.json, but when trying to use a variable name containing an emoji in a directive parameter I get:

> ⇨ Accumulated: Unexpected: Error with parameters expansion for node 'server.rudder.local' (root); cause was: Accumulated: When processing directive 'File content'; cause was: Accumulated: On variable 'GENERIC_FILE_CONTENT_PAYLOAD':; cause was: Accumulated: Unexpected: Error when parsing value "${node.properties[emo👋ji]}", error message is:

We can keep UTF-8, but what I fear is that a lot of things will break in unexpected ways.

Reply:
OK, what shoud we authorize? `[\-_0-9a-zA-Z]`? `~` ? All ascii without control characters?